interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -206,6 +275,41 @@ func (udb *userPromptDB) remove(id prompting.IDType) (*Prompt, error) {
 	return prompt, nil
 }
 
+// timeoutCallback is the function which should be called when the expiration
+// timer for the receiving user prompt DB expires. This method should never be
+// called directly outside of a `time.AfterFunc` call which initializes the
+// timer for the user prompt DB when it is first created.
+func (udb *userPromptDB) timeoutCallback(pdb *PromptDB, user uint32) {
+	pdb.mutex.Lock()
+	if pdb.maxIDMmap.IsClosed() {
+		return
```

Comment:
needs pdb.mutex.Unlock()?

Reply:
Yes, I think I missed the defer unlock. Thanks for catching it!

Reply:
Ah I see, yes, can't defer here because we don't want to hold the lock while recording the notice, so we're not blocked waiting for the state lock. The function call will still not return until state lock is acquired and the notice is recorded, but at least the prompting backend will be able to receive listener requests and merge them with outstanding prompts while we're waiting.